WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingIPPUsersRefresher.swift

```diff
@@ -5,41 +5,26 @@ import Storage
 /// Refreshes the CPP onboarding state if there are IPP transactions stored
 ///
 class CardPresentPaymentsOnboardingIPPUsersRefresher {
-    private let storageManager: StorageManagerType
+    private let stores: StoresManager
     private let cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCaseProtocol
 
-    private lazy var ordersResultsController: ResultsController<StorageOrder> = {
-        return ResultsController<StorageOrder>(storageManager: storageManager, sortedBy: [])
-    }()
-
-    init(storageManager: StorageManagerType = ServiceLocator.storageManager,
+    init(stores: StoresManager = ServiceLocator.stores,
          cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCaseProtocol = CardPresentPaymentsOnboardingUseCase()) {
-        self.storageManager = storageManager
+        self.stores = stores
         self.cardPresentPaymentsOnboardingUseCase = cardPresentPaymentsOnboardingUseCase
-
-        try? ordersResultsController.performFetch()
     }
 
     func refreshIPPUsersOnboardingState() {
```

Comment:
Should this be called when we switch sites? I'm not sure, but I think so and I can't see it happening at the moment. I see the cache being invalidated though.

Reply:
After switching sites, I took a payment on my WCPay and Stripe (using TTP), and then went to Manage card readers.

When it checked the cache, it showed these:
```
(lldb) po state
▿ CardPresentPaymentOnboardingState
  ▿ completed : 1 element
    ▿ plugin : CardPresentPaymentsPluginState
      - preferred : Yosemite.CardPresentPaymentsPlugin.stripe
      ▿ available : 1 element
        - 0 : Yosemite.CardPresentPaymentsPlugin.stripe

(lldb) po cachedValue
▿ CardPresentPaymentOnboardingState
  ▿ completed : 1 element
    ▿ plugin : CardPresentPaymentsPluginState
      - preferred : Yosemite.CardPresentPaymentsPlugin.wcPay
      ▿ available : 2 elements
        - 0 : Yosemite.CardPresentPaymentsPlugin.wcPay
        - 1 : Yosemite.CardPresentPaymentsPlugin.stripe
```

Which looks potentially like it still had the cached value from the previous site, which was Stripe-only.

It happened a couple of times, even after it had set the state correctly, so this may not be caused by this PR and instead be a leak where we have multiple subscriptions to the card reader publisher living on with some captured state, or something like that. It's weird though, and I thought worth noting. This was found [using a breakpoint at `CardPresentPaymentsOnboardingUseCase.swift:refreshIfNecessary()`](https://github.com/woocommerce/woocommerce-ios/blob/f2e64b606cb13c614ab493330bdd968e07b7a18f/WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person%20Payments/CardPresentPaymentsOnboardingUseCase.swift#L88)

Reply:
As we talked, we invalidate the cache after switching sites, which in my opinion should be enough, given the unlikelihood of the switching sites scenario. Once the cache is invalidated the onboarding will run before the payment as usual.